sound/soc/sof/intel/hda-dsp.c

```diff
@@ -707,16 +703,6 @@ static int hda_resume(struct snd_sof_dev *sdev, bool runtime_resume)
 		if (sdev->system_suspend_target == SOF_SUSPEND_NONE)
 			hda_codec_jack_check(sdev);
 	}
-
-	/* turn off the links that were off before suspend */
-	list_for_each_entry(hlink, &bus->hlink_list, list) {
-		if (!hlink->ref_count)
-			snd_hdac_ext_bus_link_power_down(hlink);
-	}
```

Comment:
@plbossart this looks OK but if you look at hda_dsp_resume() we power up the links that were ON prior to suspend for S0ix and that seems to make sense. Do we need something similar here as well for S3?

Reply:
Well that's the key point. The S0ix entry/exit are perfectly balanced and logical. This S3 resume makes no sense whatsover. Either we are missing something on S3 entry or this block is just useless. I suspect it's the latter, given the logical inversion and nonsensical comment.